File: ShopAppSpring/src/main/java/com/duchung/shopappspring/dtos/TransactionDocumentDTO.java
package com.duchung.shopappspring.dtos;


import com.duchung.shopappspring.models.User;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class TransactionDocumentDTO {
    private String name;

    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("raw_products")
    List<RawProductDTO> rawProducts;
}

File: shopapp-angular/src/app/manager/dtos/transaction.document.dto.ts
import { RawProductDTO } from './raw.product.dto';

export class TransactionDocumentDTO {
  name: string;
  user_id: number;
  raw_products: RawProductDTO[];

  constructor(data: any) {
    this.name = data.name;
    this.user_id = data.user_id;
    this.raw_products = data.raw_products;
  }
}
